src/app/page.tsx

```diff
@@ -1,42 +1,54 @@
 'use client';
 
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
 import styles from '@/page.module.scss';
 import toast from 'react-hot-toast';
+import { useNewsContext } from '@/context/news.context';
 import NewsList from './components/NewsList/NewsList';
+import Spinner from './components/Spinner/Spinner';
 import getNews from './utils/getNews';
-import { NewsWithId } from './models/types';
 
 type Props = {
   searchParams: { q: string };
 };
 
 function Home({ searchParams }: Props) {
-  const [news, setNews] = useState<NewsWithId[]>([]);
+  const { news, loading, setNews, setLoading } = useNewsContext();
 
   useEffect(() => {
     (async () => {
-      if (!searchParams.q) return;
+      try {
+        setLoading(true);
 
-      const response = await getNews(searchParams.q);
+        if (!searchParams.q) {
+          setLoading(false);
```

Comment:
Please remove this `setLoading` because finally will fire before or right after return, I don’t remember exactly